internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -382,7 +382,6 @@ func resourcePostgresqlFlexibleServer() *pluginsdk.Resource {
 			}
 
 			newMb = newStorageMbRaw.(int)
-			newTier = newTierRaw.(string)
 
 			// storage_mb can only be scaled up...
```

Comment:
I assume that resourcePostgresqlFlexibleServerUpdate also needs to be updated since currently the update func wouldn't set the default value for storage_mb and storage_tier when they aren't set in the tf config. Below is the sample for the fix.

```
storage := expandArmServerStorage(d)
		var storageMb int

		if storage.StorageSizeGB == nil || *storage.StorageSizeGB == 0 {
			// set the default value for storage_mb...
			storageMb = 32768
			storage.StorageSizeGB = pointer.FromInt64(int64(32))
			log.Printf("[DEBUG]: Default 'storage_mb' Set -> %d\n", storageMb)
		} else {
			storageMb = int(*storage.StorageSizeGB) * 1024
		}

		if storage.Tier == nil || *storage.Tier == "" {
			// determine the correct default storage_tier based
			// on the defined storage_mb...
			storageTierMappings := validate.InitializeFlexibleServerStorageTierDefaults()
			storageTiers := storageTierMappings[storageMb]
			storage.Tier = pointer.To(storageTiers.DefaultTier)
			log.Printf("[DEBUG]: Default 'storage_tier' Set -> %q\n", storageTiers.DefaultTier)
		}

		storageUpdateParameters := servers.ServerForUpdate{
			Properties: &servers.ServerPropertiesForUpdate{
				Storage: storage,
			},
		}
```